server/src/main/java/org/elasticsearch/search/aggregations/bucket/global/GlobalAggregatorFactory.java

```diff
@@ -29,15 +29,15 @@ public GlobalAggregatorFactory(
     ) throws IOException {
         super(name, context, parent, subFactories, metadata);
         if (subFactories.isInSortOrderExecutionRequired()) {
-            throw new AggregationExecutionException("Time series aggregations cannot be used inside global aggregation.");
+            throw new IllegalArgumentException("Time series aggregations cannot be used inside global aggregation.");
         }
     }
 
     @Override
     public Aggregator createInternal(Aggregator parent, CardinalityUpperBound cardinality, Map<String, Object> metadata)
         throws IOException {
         if (parent != null) {
-            throw new AggregationExecutionException(
+            throw new IllegalArgumentException(
```

Comment:
Probably we can add a case to `AggregationErrors` for this kind of nesting error.  I know it happens in a few places, but I didn't refactor it in this PR because I was having a hard time getting the messages to line up.